IoTDeviceSimulator/azure-functions/d2c-simulator/index.js

```diff
@@ -143,7 +303,22 @@ module.exports = async function (context, req) {
     }, (parseInt(device.telemetrySendInterval)))
   }
 
-  // Run Simulator
+  function SetIntervalForResolve(client, device) {
```

Comment:
The functions SetIntervalForResolve, updateSensorForResolve and sendTelemetryForResolve can be eliminated and the existing functions can be reused by just modifying the signalArray for resolve condition